test/jdk/java/awt/FileDialog/MultipleMode.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2010, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Checkbox;
+import java.awt.EventQueue;
+import java.awt.FileDialog;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Panel;
+import java.awt.TextArea;
+import java.io.File;
+
+/*
+ * @test
+ * @bug 6467204
+ * @summary Need to implement "extended" native FileDialog for JFileChooser
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual MultipleMode
+*/
+
+public class MultipleMode {
+
+    private static final String INSTRUCTIONS =
+            """
+             1. Turn the 'multiple' checkbox off and press the 'open' button
+             2. Verify that the file dialog doesn't allow the multiple file selection
+             3. Select any file and close the file dialog
+             4. The results will be displayed, verify the results
+             5. Turn the 'multiple' checkbox on and press the 'open' button
+             6. Verify that the file dialog allows the multiple file selection
+             7. Select several files and close the file dialog
+             8. The results will be displayed, verify the results.
+            """;
+
+    public static void main(String[] args) throws Exception {
+
+        PassFailJFrame passFailJFrame = PassFailJFrame
+                .builder()
+                .title("MultipleMode test instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(15)
+                .columns(40)
+                .build();
+
+        EventQueue.invokeAndWait(MultipleMode::init);
```

Comment:
I suggest using `testUI` method. Make your `init` method return `Frame` after you call `pack()`, don't call `setVisible(true)`.

```java
    .testUI(MultipleMode::init)
```

Then you'll be able to chain `awaitAndCheck()` call.